thunder/tests/test_examine_memory.py

```diff
@@ -150,7 +150,8 @@ def test_nanogpt_block(executor, device, dtype):
         expected_return_calculated_mem = get_return_memory(fw_extrace.bound_symbols[-1]) - 4 * 2 * 1024 * 768
         assert expected_return_calculated_mem == sum(fw_alloc_mem[1].values())
 
-        assert bw_alloc_mem[0] == 361783296
+        # NOTE: Questionable memory bump and haven't investigated yet
+        assert bw_alloc_mem[0] == 361881600
```

Comment:
This change seems to be strange. Note that I need to further investigate this.

Reply:
I noticed that in the bwd trace, we are seeing number proxies showing up in the trace. I'm imagining that's the reason of the bump in analysis.

My naive understanding is that, currently since we are not registering numbers as proxies. number proxies should not show up in trace through `thunder.jit` and operations should still back in. I'm not totally sure why backward is seeing these.

Reply:
But the problem with this bump is actually in an change in `nvfuserex_impl.py`. where we are rejecting `torch.sum` with a reduction axes as NumberProxies.

Somehow after grad transform, the trace is taking a ProxyNumber as the reduction axis and the added logic is rejecting that. So the change in behavior is a right thing to do. But do we understand why grad transform is doing that?

I get this bwd trace in this test from running our main branch
```
  t411 = torch.sum(t409, i51, True, dtype=None)  # t411: "cuda:0 f32[2, 12, 1024, 1]"
    # t411 = ltorch.sum(t409, i51, True, dtype=None)  # t411: "cuda:0 f32[2, 12, 1024, 1]"
      # t552 = prims.sum(t409, (3,))  # t552: "cuda:0 f32[2, 12, 1024]"
      # t411 = prims.broadcast_in_dim(t552, [2, 12, 1024, 1], [0, 1, 2])  # t411: "cuda:0 f32[2, 12, 1024, 1]"
  del t409, i51
```

```
commit d6237c88d26aa4a5541f3ad2e955140264afb068 (HEAD -> main, origin/main, origin/HEAD)
Author: Ivan Yashchuk <IvanYashchuk@users.noreply.github.com>
Date:   Fri May 10 18:26:46 2024 +0300

    Fix flaky test_prim_inplace_copy_fwd (#399)
```

Reply:
I'll open an issue to track this.

Reply:
Following up with #403